Do not include two directories with generated gnulib headers in -I flags,
or #include_next directive for error.h fails.

diff -urN gettext-0.22.3/gettext-tools/src/Makefile.am.orig gettext-0.22.3/gettext-tools/src/Makefile.am
--- gettext-0.22.3/gettext-tools/src/Makefile.am.orig	2023-10-27 20:04:58.982538600 +0200
+++ gettext-0.22.3/gettext-tools/src/Makefile.am	2023-10-27 20:40:13.497536600 +0200
@@ -307,7 +307,7 @@
 
 # Use the dummy libtextstyle from gnulib, as libgettextpo does.
 LT_LIBTEXTSTYLE =
-AM_CPPFLAGS += -I../libgettextpo -I$(top_srcdir)/libgettextpo
+AM_CPPFLAGS += -I$(top_srcdir)/libgettextpo
 
 # How to build libgettextsrc.la.
 # Need ../gnulib-lib/libgettextlib.la.
diff -urN gettext-0.22.3/gettext-tools/libgettextpo/Makefile.gnulib.orig gettext-0.22.3/gettext-tools/libgettextpo/Makefile.gnulib
--- gettext-0.22.3/gettext-tools/libgettextpo/Makefile.gnulib.orig	2023-10-02 23:07:33.000000000 +0200
+++ gettext-0.22.3/gettext-tools/libgettextpo/Makefile.gnulib	2023-10-27 21:08:15.039903500 +0200
@@ -751,11 +751,12 @@
 # We need the following in order to create a dummy placeholder for
 # <textstyle.h>.
 textstyle.h: textstyle.in.h $(top_builddir)/config.status
-	$(AM_V_GEN)rm -f $@-t $@ && \
+	$(AM_V_GEN)rm -f $@-t $@ ../gnulib-lib/$@ && \
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/textstyle.in.h; \
 	} > $@-t && \
-	mv $@-t $@
+	mv $@-t $@ && \
+	cp $@ ../gnulib-lib/$@
 MOSTLYCLEANFILES += textstyle.h textstyle.h-t
 
 EXTRA_DIST += textstyle.in.h
